### PR TITLE
Docs building docs and config docs

### PR DESCRIPTION
**Pull request by dihm**
Finally finished a pass at some detailed documentation for how to build the documentation locally. Also went ahead and added some specific stuff for how to configure a new installation of the suite.

There is a lot going on here, and I'm completely open to suggestions for how to improve this. I consider this the last big doc dump I'll worry about for a good long while (hopefully).

### CODE REVIEW

**Comment by dihm** [comment]
Added a small page on common trouble shooting steps I find I have to tell everyone when they first start using labscript. It may be a good idea to add such a page to many of the repos, just as a way of helping new users avoid the frustration of common pitfalls.

File: docs/source/configuration.rst [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+At the end of the file, you will call the **labscript** functions `start` and `stop`, without any actual instructions commanded to any device in the shot.
+This file must be saved as the name defined and location specified by the `connection_table_py` key of the labconfig file.
+The defaul location is in the `apparatus_name/labscriptlib` sub-folder of the userlib.
+
+A very simple connection table that defines a PrawnBlaster pseudoclock and an NI DAQ with two named channels is as follows.
+
+.. code-block:: python
+
+   from labscript import *
+
```
Comment (zakv) [inline docs/source/configuration.rst:77]:
```suggestion

    # Import the classes needed for the devices which will be used.
```

File: docs/source/configuration.rst [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+   from labscript_devices.PrawnBlaster.labscript_devices import PrawnBlaster
+   from labscript_devices.NI_DAQmx.models.NI_USB_6363 import NI_USB_6363
+
```
Comment (zakv) [inline docs/source/configuration.rst:80]:
```suggestion

    # Add a Prawnblaster named 'prawn' which will be the pseudoclock that triggers the other devices.
```

Maybe this comment should be wrapped to keep the line length down?

File: docs/source/configuration.rst [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+   PrawnBlaster(name='prawn', com_port='COM6', num_pseudoclocks=1)
+
```
Comment (zakv) [inline docs/source/configuration.rst:82]:
```suggestion

    # Add an NI USB-6363 multifunction I/O device.
```

File: docs/source/configuration.rst [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+   NI_USB_6363(name='daq', MAX_name='Dev1',
+               parent_device=prawn.clocklines[0], clock_terminal='/Dev1/PFI0',
+               acquisition_rate=100e3)
+
```
Comment (zakv) [inline docs/source/configuration.rst:86]:
```suggestion

    # Add output channels for the NI USB-6363.
```

File: docs/source/configuration.rst [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+   AnalogOut('ao0', daq, 'ao0')
+   AnalogOut('ao1', daq, 'ao1')
+
```
Comment (zakv) [inline docs/source/configuration.rst:89]:
```suggestion

    # The rest is boilerplate syntax which is the same for any apparatus.
```

**Review by zakv — APPROVED** [review]
This looks great! I have a few minor comments/suggestions; feel free to incorporate or ignore them!

1. There are two parts of the docs that describe the PR submission process: "Committing Docs" and "Contributing to the labscript suite". I think it would be a good idea to consolidate those instructions in one place and then have one page link to the other. I think it would make most sense to put them in "Contributing to the labscript suite" but the instructions currently in "Committing Docs" are much more comprehensive. My suggestion would be to move the instructions from "Committing Docs" to "Contributing to the labscript suite" then link there from "Committing Docs."
2. I think it would be helpful to add in-code comments to the example connection table in "Configuring the labscript suite for first run." I've added suggested comments, though feel free to edit them.
3. I believe this incorporates and/or supersedes #57 right? If so that PR can be closed when this one is merged.

**Comment by dihm** [comment]
These are all good suggestions. I've gone ahead and added all the inline comments. I figured the duplicated PR instructions were a bit much. I've gone ahead and consolidated. Hopefully experienced coders don't feel condescended to.

And yes, this does largely supersede #57, so I'll close it once this is ready to go.

**Pull request merged**